packages/devtools_app/lib/src/screens/vm_developer/vm_class_display.dart

```diff
@@ -20,64 +25,122 @@ class VmClassDisplay extends StatelessWidget {
 
   @override
   Widget build(BuildContext context) {
-    return Column(
+    return Row(
       children: [
-        Flexible(
-          flex: 5,
-          child: ClassInfoWidget(
-            clazz: clazz,
+        Expanded(
+          child: Column(
+            children: [
+              ClassInfoWidget(
+                clazz: clazz,
+              ),
+              Flexible(
+                child: ListView(
+                  children: [
+                    RetainingPathWidget(
+                      fetching: clazz.fetchingRetainingPath,
+                      retainingPath: clazz.retainingPath,
+                      onExpanded: _onExpandRetainingPath,
+                    ),
+                    InboundReferencesWidget(
+                      fetching: clazz.fetchingInboundRefs,
+                      inboundReferences: clazz.inboundReferences,
+                      onExpanded: _onExpandInboundRefs,
+                    )
+                  ],
+                ),
+              ),
+            ],
           ),
         ),
-        Flexible(
-          flex: 4,
-          child: ClassInstancesWidget(
-            instances: clazz.instances,
-          ),
-        )
+        if (displayClassInstances)
+          Flexible(
+            child: ClassInstancesWidget(
+              instances: clazz.instances,
+            ),
+          )
       ],
     );
   }
+
+  void _onExpandRetainingPath(bool expanded) {
+    if (clazz.retainingPath.value == null) clazz.requestRetainingPath();
+  }
+
+  void _onExpandInboundRefs(bool expanded) {
+    if (clazz.inboundReferences.value == null) clazz.requestInboundsRefs();
+  }
 }
 
 /// Displays general VM information of the Class Object.
-class ClassInfoWidget extends StatelessWidget {
+class ClassInfoWidget extends StatelessWidget implements PreferredSizeWidget {
   const ClassInfoWidget({
     required this.clazz,
   });
 
   final ClassObject clazz;
 
+  // TODO: change this value if adding/removing rows.
+  static const numberOfRows = 9;
+
   @override
   Widget build(BuildContext context) {
-    return VMInfoCard(
-      title: 'General Information',
-      rowKeyValues: [
-        MapEntry('Object Class', clazz.obj.type),
-        MapEntry(
-          'Shallow Size',
-          prettyPrintBytes(
-            clazz.obj.size ?? 0,
-            includeUnit: true,
-            kbFractionDigits: 3,
+    return SizedBox.fromSize(
+      size: preferredSize,
+      child: VMInfoCard(
+        title: 'General Information',
+        rowKeyValues: [
+          MapEntry('Object Class', clazz.obj.type),
```

Comment:
Can we lift this list into a local variable in `build` and then explicitly check that the number of rows matches `numberOfRows`, throwing a `StateError` if they don't match?

Reply:
Done. I used the length of the list instead, to obtain the number of rows.